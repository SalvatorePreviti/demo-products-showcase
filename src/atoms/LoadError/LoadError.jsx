import React from 'react'

function LoadError() {
  return (
    <div>
      <b>Loading error.</b> Try to reload
    </div>
  )
}

export default LoadError
