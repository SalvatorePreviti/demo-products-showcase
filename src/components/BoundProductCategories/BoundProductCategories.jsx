import React from 'react'
import { connect } from 'react-redux'
import ProductCategories from '../../atoms/ProductCategories/ProductCategories'
import actionSelectCategory from '../../domain/actions/actionSelectCategory'

function mapStateToProps(state) {
  const { categories, activeCategory } = state
  return { categories, activeCategory }
}

function mapDispatchToProps(dispatch) {
  return {
    onSelectCategory(category) {
      dispatch(actionSelectCategory(category))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(React.memo(ProductCategories))
