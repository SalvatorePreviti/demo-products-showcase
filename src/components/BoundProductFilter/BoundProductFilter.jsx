import React from 'react'
import { connect } from 'react-redux'
import ProductFilter from '../../atoms/ProductFilter/ProductFilter'
import actionSetProductFilter from '../../domain/actions/actionSetProductFilter'

function mapStateToProps(state) {
  const { productFilter } = state
  return { productFilter }
}

function mapDispatchToProps(dispatch) {
  return {
    onChange(filter) {
      dispatch(actionSetProductFilter(filter))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(React.memo(ProductFilter))
