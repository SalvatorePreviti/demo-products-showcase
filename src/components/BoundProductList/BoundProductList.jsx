import React from 'react'
import { connect } from 'react-redux'
import ProductList from '../../atoms/ProductList/ProductList'
import actionToggleProduct from '../../domain/actions/actionToggleProduct'

function mapStateToProps(state) {
  const { filteredProducts, activeProductsById } = state
  return {
    items: filteredProducts,
    activeProductsById: activeProductsById
  }
}

function mapDispatchToProps(dispatch) {
  return {
    onSelect(product) {
      dispatch(actionToggleProduct(product))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(React.memo(ProductList))
