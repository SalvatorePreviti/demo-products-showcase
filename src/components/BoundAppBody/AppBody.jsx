import React from 'react'
import PropTypes from 'prop-types'

import LoadError from '../../atoms/LoadError/LoadError'
import Loading from '../../atoms/Loading/Loading'
import BoundProductCategories from '../BoundProductCategories/BoundProductCategories'
import BoundProductFilter from '../BoundProductFilter/BoundProductFilter'
import BoundCategoryProductList from '../BoundCategoryProductList/BoundCategoryProductList'

function AppBody(props) {
  const { loading, loadFailed } = props

  if (loadFailed) {
    return <LoadError />
  }

  if (loading) {
    return <Loading />
  }

  return (
    <div>
      <BoundProductCategories />
      <BoundProductFilter />
      <BoundCategoryProductList />
    </div>
  )
}

AppBody.propTypes = {
  loading: PropTypes.bool,
  loadFailed: PropTypes.bool
}

export default AppBody
