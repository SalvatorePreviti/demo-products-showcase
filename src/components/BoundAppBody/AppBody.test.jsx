import React from 'react'
import { shallow } from 'enzyme'

import AppBody from './AppBody'
import LoadError from '../../atoms/LoadError/LoadError'
import Loading from '../../atoms/Loading/Loading'
import BoundProductCategories from '../BoundProductCategories/BoundProductCategories'
import BoundProductFilter from '../BoundProductFilter/BoundProductFilter'
import BoundProductList from '../BoundProductList/BoundProductList'

describe('AppBody', () => {
  describe('with load error', () => {
    let root

    beforeAll(() => {
      root = shallow(<AppBody loadFailed={true} />)
    })

    it('snapshots', () => {
      expect(root).toMatchSnapshot()
    })

    it('renders a LoadError', () => {
      expect(root.find(LoadError)).toHaveLength(1)
    })

    it('does not render anything else', () => {
      expect(root.children()).toHaveLength(0)
    })
  })

  describe('with loading', () => {
    let root

    beforeAll(() => {
      root = shallow(<AppBody loading={true} />)
    })

    it('snapshots', () => {
      expect(root).toMatchSnapshot()
    })

    it('renders a Loading', () => {
      expect(root.find(Loading)).toHaveLength(1)
    })

    it('does not render anything else', () => {
      expect(root.children()).toHaveLength(0)
    })
  })

  describe('when ready', () => {
    let root

    beforeAll(() => {
      root = shallow(<AppBody />)
    })

    it('snapshots', () => {
      expect(root).toMatchSnapshot()
    })

    it('Renders BoundProductCategories', () => {
      expect(root.find(BoundProductCategories)).toHaveLength(1)
    })

    it('Renders BoundProductFilter', () => {
      expect(root.find(BoundProductFilter)).toHaveLength(1)
    })

    it('Renders BoundProductList', () => {
      expect(root.find(BoundProductList)).toHaveLength(1)
    })
  })
})
