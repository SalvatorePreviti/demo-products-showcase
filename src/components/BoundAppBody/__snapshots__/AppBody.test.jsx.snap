// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBody when ready snapshots 1`] = `
<div>
  <Connect(Component) />
  <Connect(Component) />
  <Connect(Component) />
</div>
`;

exports[`AppBody with load error snapshots 1`] = `<LoadError />`;

exports[`AppBody with loading snapshots 1`] = `<Loading />`;
